### PR TITLE
[IOS-0014] Fix multiple next page calls

### PR DESCRIPTION
**Pull request by gabrielpc4**

### CODE REVIEW

File: MoviesApp/Presentation/Scenes/MoviesList/MoviesListViewModel.swift [hunk 1]
```diff
@@ -6,6 +6,7 @@ final class MoviesListViewModel {
     private let lastPage = 500
     private(set) var paginaAtual: Int = 1
     private let getMoviesUseCase: GetMoviesUseCaseProtocol
+    var isFetchingNextPage: Bool = false
```
Comment (RicardoBC89) [inline MoviesApp/Presentation/Scenes/MoviesList/MoviesListViewModel.swift:9]: Let it be private
```suggestion
    private var isFetchingNextPage: Bool = false
```

**Review by RicardoBC89 — CHANGES_REQUESTED** [review]

File: MoviesApp/Presentation/Scenes/MoviesList/MoviesListViewModel.swift [hunk 2]
```diff
@@ -14,7 +15,9 @@ final class MoviesListViewModel {
     func fetchMovies(pagina: Int) {
         guard pagina <= lastPage else { return }
         isLoading.value = true
+        isFetchingNextPage = true
         getMoviesUseCase.execute(pagina: pagina, viewModelCompletionHandler: { [weak self] movies, error in
+            isFetchingNextPage = false
```
Comment (RicardoBC89) [inline MoviesApp/Presentation/Scenes/MoviesList/MoviesListViewModel.swift:20]: Are your sure this doesn't make a memory leak????
```suggestion
            self?.isFetchingNextPage = false
```

File: MoviesApp/Presentation/Scenes/MoviesList/MoviesListViewModel.swift [hunk 3]
```diff
@@ -24,7 +27,10 @@ final class MoviesListViewModel {
     }
 
     func nextPage() {
-        paginaAtual += 1
-        fetchMovies(pagina: paginaAtual)
+        if !isFetchingNextPage {
+            paginaAtual += 1
+            fetchMovies(pagina: paginaAtual)
+        }
+
```
Comment (RicardoBC89) [inline MoviesApp/Presentation/Scenes/MoviesList/MoviesListViewModel.swift:34]: Delete the line.
Please, pay attention, this is a importante code

File: MoviesApp/Presentation/Scenes/MoviesList/MoviesListViewModel.swift (hunk 2 above)
Comment (gabrielpc4) [reply MoviesApp/Presentation/Scenes/MoviesList/MoviesListViewModel.swift:20]: > 

Thank you for catching that. Fixing it!

**Review by RicardoBC89 — APPROVED** [review]

**Pull request merged**